flyteplugins/go/tasks/pluginmachinery/flytek8s/pod_helper.go

```diff
@@ -348,6 +348,15 @@ func ApplyFlytePodConfiguration(ctx context.Context, tCtx pluginsCore.TaskExecut
 		IncludeConsoleURL: hasExternalLinkType(taskTemplate),
 	}
 
+	// iterate over the initContainers first
+	for index := range podSpec.InitContainers {
```

Comment:
lgtm, could we add a test [here](https://github.com/flyteorg/flyte/blob/7d788cbe1e3f395d0e6ad862c4fb310458d09dbf/flyteplugins/go/tasks/plugins/k8s/pod/container_test.go#L169)?